src/program/snabbvmx/doc/README.troubleshooting.md

```diff
@@ -320,10 +320,37 @@ NOTE: Currently the test is not working correctly: the returned MAC should be
 `02:99:99:99:99:99`.
 
 
-## End-to-end tests - details
+## Crafting and running end-to-end tests
+
+### Brief steps
+
+(1) run the interactive check with "-r" param to derive the counters and out.pcaps
+
+`snabb snabbvmx check -r  ./CONF.cfg "./V4-IN.PCAP" "./V6-IN.PCAP" "./outv4.pcap" "./outv6.pcap" COUNTERS.lua`
+
+(2) place derived counters.lua in "snabb/src/program/snabbvmx/tests/end-to-end/data/counters"
+
+(3) place derived and expected out.pcaps in "snabb/src/program/snabbvmx/tests/end-to-end/data"
+
+(4) edit the "test_env.sh" in snabb/src/program/snabbvmx/tests/end-to-end to have the test scripted
+
+(5) run the scripted test:
+
+`snabb/src/program/snabbvmx/tests/end-to-end$ sudo ./end-to-end.sh`
+
+
+### How the system test works in detail
+
+The way the test system works is by passing the input IPv4/IPv6 packets (via pre-recorded pcap-files) to the lwAFTR and comparing the expected packet output (pcap-file) to the packet output that the lwAFTR has generated.
+The optional counters file is compared too to the actual counters file obtained after running the test. This is beeing reflected in snabbvmx check syntax:
+
+`CONF V4-IN.PCAP V6-IN.PCAP V4-OUT.PCAP V6-OUT.PCAP [COUNTERS.LUA]`
+
+V4-OUT.PCAP, V6-OUT.PCAP and COUNTERS.LUA are expected output. These output is compared to files stored temporarily in /tmp/endoutv4.pcap, /tmp/endoutv6.pcap and /tmp/counters.lua. These temporal files are the actual output produced by the lwAFTR after running a test. In order for a test to pass, the actual output must match the expected output. So it's not only that the counters file should match, but also the output .pcap files. (read: if a counters.lua file is provided, then it must still match the V4-OUT.PCAP and V6-OUT.PCAP)
```

Comment:
`expected output` -> `expected outputs`
`These output is compared` -> `These outputs are compared`
`These temporal files` -> `These temporary files`